src/sections/messages/filter.js

```diff
@@ -0,0 +1,7 @@
+export default function filter(query, data) {
+    const newQuery = query.replace(/[^a-z0-9]/gi, '').split('').join('.*');
+    return data.filter((item) => {
+        const regex = new RegExp(newQuery, 'i');
+        return regex.test(item.name);
+    });
+}
```

Comment:
this is twice so create a utils folder in src and put it in a file named searchFilter.js and remove from here and component section